typing/typecore.ml

```diff
@@ -5100,6 +5100,7 @@ and type_let ?check ?check_strict
 
   let (pat_list, exp_list, new_env, unpacks, _pvs) =
     with_local_level begin fun () ->
+      if existential_context = At_toplevel then Typetexp.TyVarEnv.reset ();
```

Comment:
This is new, what is its purpose?

Reply:
This is the former call to `TyVarEnv.reset` from `type_binding` (identified by `existential_context=At_toplevel` which is only true coming from this path).

Reply:
Makes sense, thanks!